### ISSUE TITLE
Catboost can not load large model

### ISSUE THREAD

**Issue by fenqingr** [catboost/catboost#878]
Problem: Catboost can save the model using catboost.save_model but can not load it back
catboost version: catboost-R-Windows-0.15.1.tgz
Operating System:OS Name	Microsoft Windows 10 Pro,Version	10.0.18362 Build 18362
CPU: i9-9900K
GPU: GTX1080Ti
R version: R version 3.5.3 (2019-03-11) -- "Great Truth"

Below is the code:

devtools::install_url('https://github.com/catboost/catboost/releases/download/v0.15.1/catboost-R-Windows-0.15.1.tgz',args = c("--no-multiarch"))
library(catboost)

set.seed(10)
X=data.frame(matrix(rnorm(1000000*50),ncol=50))
Y=apply(X,1,function(x) max(x)/sum(abs(x)))
pool=catboost.load_pool(X,label =Y)
fit_params <- list(iterations = 3000,
                   loss_function = 'RMSE',
                   depth =16,
                   learning_rate = 0.005,
                   task_type = 'GPU')

model <- catboost.train(pool, params = fit_params)

catboost.save_model(model,"D:/RData/test catboost model save.RData")
model.loaded=catboost.load_model("D:/RData/test catboost model save.RData")

The model was successful saved
> catboost.save_model(model,"D:/RData/test catboost model save.RData")
[1] TRUE
But can't be load back . I got the error:
> model.loaded=catboost.load_model("D:/RData/test catboost model save.RData")
Error in catboost.load_model("D:/RData/test catboost model save.RData") : 
  c:/goagent/pipelines/buildmaster/catboost.git/catboost/libs/model/model.cpp:861: Flatbuffers model verification failed.

I found this issue will not appear if I set the depth=6; however, if I set it to 16 it will appear.

Please investigate. Thanks a lot!

**Comment by annaveronika** [comment]
One symmetric tree in the model with depth 16 has 2^16 leafs.
3000 * 2^16 leafs, each leaf is a double, thus it is 3000 * 2^16 * 8 bytes just for storing the leaf values (we also store weights in leafs). Are you sure you want this model?

We will look into the failure though, thanks for the report!

**Comment by fenqingr** [comment]
Hi Annaveronika,

Thank for the quick reply! Actually the model is for finical data which has very weak association between predictors and label. (r-square is about 0.003). My experiment using lightgbm shown depth more than 16 is beneficial to the predicting power then those small trees. 

Thank you for the support again!

**Comment by annaveronika** [comment]
LightGBM uses not symmetric trees by default that are grown node by node, not level by level.
To build such trees in CatBoost you need to set grow_policy to Lossguide and use GPU training (if you really want this type of trees)
Here are some tips about grow_policy https://catboost.ai/docs/concepts/parameter-tuning.html#symmetric-trees

**Comment by fenqingr** [comment]
Great to know! Thank you for the explain. Hope your team can fix this issue soon; anyway, it is a bug that deserve a fix. Many thanks!

**Comment by annaveronika** [comment]
We've looked into the failure, here is the issue with the plan and explanation: https://github.com/catboost/catboost/issues/842 we'll add option validation.

Closing this one as a duplicate.
